app/jobs/destroy_split_pages_job.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class DestroySplitPagesJob < ApplicationJob
+  queue_as :default
+
+  def perform(id)
+    # TODO: Make work with Valkyrie
+    work = ActiveFedora::Base.where(id: id).first
```

Comment:
does this need to become something like..?
```suggestion
    work = Hyrax.query_service.find_by(id: id)
```

Reply:
This will require further rework.  The job is very much dependent on ActiveFedora.

Reply:
Wondering if the following is a suitable replacement:

```ruby
class DestroySplitPagesJob < ApplicationJob
  queue_as :default

  def perform(id)
    work = nil
    begin
      work = Hyrax.query_service.find_by(id: id)
    rescue Valkyrie::Persistence::ObjectNotFoundError
      return
    end

    return unless work.is_child

    # Does this work?
    transaction = Hyrax::Transactions::WorkDestroy.new
    transaction.with_step_args.call(work)
  end
end
```

Reply:
I'm going to test out with a spec tomorrow morning:

```
# frozen_string_literal: true

require 'spec_helper'

RSpec.describe DestroySplitPagesJob do
  describe '#perform' do
    let(:work) { FactoryBot.valkyrie_create(:generic_work_resource, :with_file_and_work) }

    it "deletes the work, child works, and file sets" do
      described_class.perform_now(work.id)
      # MAKE IT SO
    end
  end
end
```

Reply:
Hmm…my spec encountered the following:

<img width="767" alt="Screenshot 2024-04-18 at 5 16 33 PM" src="https://github.com/samvera/hyku/assets/2130/7371257c-838b-452a-901c-786882ecb735">

Reply:
Here's the spec I have that is finding another problem:

```ruby
# frozen_string_literal: true

require 'spec_helper'

RSpec.describe DestroySplitPagesJob do
  describe '#perform' do
    let(:work) { FactoryBot.valkyrie_create(:generic_work_resource, :with_file_and_work, is_child: true) }

    it "deletes the work, child works, and file sets" do
      # Ensure that we have the expected members: a file set and a work
      members = work.members
      child_work = members.detect(&:work?)
      file_set = members.detect(&:file_set?)
      expect(child_work).to be_present
      expect(file_set).to be_present

      # When we raise an exception within a job, that exception is returned.
      result = described_class.perform_now(work.id.to_s)

      # Hence we need to check if we raised an exception within the job.
      expect(result).not_to be_a(Exception)

      # Assuming everything's working, we should no longer be able to find the work
      expect { Hyrax.query_service.find_by(id: work.id) }.to raise_error Valkyrie::Persistence::ObjectNotFoundError

      # Nor the child work
      expect { Hyrax.query_service.find_by(id: child_work.id) }.to raise_error Valkyrie::Persistence::ObjectNotFoundError

      # Nor the file set
      expect { Hyrax.query_service.find_by(id: file_set.id) }.to raise_error Valkyrie::Persistence::ObjectNotFoundError
    end
  end
end
```

Reply:
Resolved and pushed up changes.